### ISSUE TITLE
journalctl --identifier / SYSLOG_IDENTIFIER won't match patterns

### ISSUE THREAD

**Issue by mvgrimes** [systemd/systemd#1256]
journalctl doesn't match patterns on SYSLOG_IDENTIFIER or via --identifier flag:

```
$ journalctl -t sshd | wc -l
987
```

but if patterns are used (either regex or glob): 

```
$ journalctl -t 'ssh*'
-- No Entries --
$ journalctl -t 'ssh.*'
-- No Entries --
$ journalctl SYSLOG_IDENTIFIER=ssh\*
-- No Entries --
$ journalctl SYSLOG_IDENTIFIER=ssh.
-- No Entries --
```

The man page documents patterns for that flag:

```
$ man journalctl
....
-t, --identifier=SYSLOG_IDENTIFIER|PATTERN
     Show messages for the specified syslog identifier SYSLOG_IDENTIFIER,
     or for any of the messages with a "SYSLOG_IDENTIFIER" matched by PATTERN.
```

Tested on ArchLinux:

```
$ journalctl --version
systemd 225
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP
+GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD +IDN
```

**Comment by poettering** [comment]
I figure that was a copy and paste error in the man page.

**Comment by poettering** [comment]
Fix is in #1452.
